src/app/notebook/notebook-markdown.component.ts

```diff
@@ -0,0 +1,164 @@
+import { DOCUMENT } from '@angular/common';
+import { Component, Inject, Input, OnChanges, SecurityContext } from '@angular/core';
+import { DomSanitizer, SafeHtml } from '@angular/platform-browser';
+import { MarkdownWrapperService } from '../shared/markdown-wrapper/markdown-wrapper.service';
+import { join, replaceAll, selectBestFromMimeBundle } from './notebook-helpers';
+import { Attachments, Cell } from './notebook-types';
+
+@Component({
+  selector: 'app-notebook-markdown',
+  templateUrl: './notebook-markdown.component.html',
+})
+export class NotebookMarkdownComponent implements OnChanges {
+  @Input() cell: Cell;
+  @Input() baseUrl: string;
+  html: string | SafeHtml;
+  private domParser: DOMParser = new DOMParser();
+
+  constructor(
+    private markdownWrapperService: MarkdownWrapperService,
+    private sanitizer: DomSanitizer,
+    @Inject(DOCUMENT) private document: Document
+  ) {}
+
+  ngOnChanges(): void {
+    // BE VERY CAREFUL when modifying this function, because to accomodate MathJax markup,
+    // which is destroyed by the Angular sanitizer, it must implement its own sanitization scheme.
+    const dangerousHtml = this.compileMarkdown(join(this.cell?.source), this.cell?.attachments ?? {});
+
+    // Sanitize using both the markdown wrapper and Angular sanitizers.
+    const sanitizedHtml = this.sanitize(dangerousHtml);
+
+    // Format embedded TeX math expressions.
+    const mathjaxedSanitizedHtml = this.formatMath(sanitizedHtml);
+
+    // Mark as pre-sanitized and assign to html field.
+    this.html = this.sanitizer.bypassSecurityTrustHtml(mathjaxedSanitizedHtml);
+  }
+
+  sanitize(dangerousHtml: string): string {
+    return this.sanitizer.sanitize(SecurityContext.HTML, this.markdownWrapperService.customSanitize(dangerousHtml));
+  }
+
+  compileMarkdown(markdown: string, attachments: Attachments): string {
+    // Adjustment the markdown to support attached notebook images and embedded TeX math expressions.
+    let adjusted = markdown;
+    adjusted = this.convertBackslashedDollars(adjusted);
+    adjusted = this.inlineAttachedImages(adjusted, attachments);
+    adjusted = this.preprocessLatexMath(adjusted);
+    // Convert the adjusted markdown to HTML using our wrapper.
+    return this.markdownWrapperService.customCompile(adjusted, this.baseUrl);
+  }
+
+  formatMath(html: string): string {
+    // Format embedded math using MathJax: https://docs.mathjax.org/en/latest/web/typeset.html
+    // Find the global MathJax object.
+    const mathjax = (<any>this.document?.defaultView)?.MathJax;
+    if (mathjax == undefined) {
```

Comment:
`==` -> `===`

Reply:
`mathjax == undefined`
is a shorthand for
`mathjax === undefined || mathjax === null`
Is it ok in this context, it's certain more readable.

Reply:
I changed all instances of `==` and `!=`, because at this point, code clarity is less important than getting this PR merged.  However, we should consider modifying the style guide to allow the shorthand.

Reply:
> `mathjax == undefined` is a shorthand for `mathjax === undefined || mathjax === null` Similarly, `!=` Are they ok in this context? They're certainly cleaner and more difficult to break.

Interesting, I'd never known you could do that, although it makes sense now that you've described it. I usually do `if (!mathjax)` for this situation.